### PR TITLE
chore(updatecli): add conditions for azure gallery

### PR DESCRIPTION
**Pull request by smerle33**
as per https://github.com/jenkins-infra/helpdesk/issues/3471
add condition on updating azure gallery

### CODE REVIEW

**Review by dduportal — CHANGES_REQUESTED** [review]
The current builds errors the conditions with the following message:

```
CONDITIONS:
===========

checkifazureimagejenkins-agent-ubuntu-22.04-amd64isavailable
------------------------------------------------------------
The shell 🐚 command "/bin/sh /tmp/updatecli/bin/eadb3a3fe80073b5e477a667ca48ae0e44982c048129e063f7510d77cdfcc849.sh" exited on error (exit code 1) with the following output:
----
----

command stderr output was:
----
ERROR: Please run 'az login' to setup account.

----
shell command failed. Expected exit code 0 but got 1
  => expected condition result false, got false

checkifazureimagejenkins-agent-ubuntu-22.04-arm64isavailable
------------------------------------------------------------
The shell 🐚 command "/bin/sh /tmp/updatecli/bin/97bf126ec1cd0d562fad66de676452a0a182a40ba803da01f0c49fb99ce0db55.sh" ran successfully with the following output:
----
--output tsv
----

checkifazureimagejenkins-agent-windows-2019-amd64isavailable
------------------------------------------------------------
The shell 🐚 command "/bin/sh /tmp/updatecli/bin/270e1e9de8a6e36d2090d7dd738991d63c05072cd0a843fdf1496c3d90375986.sh" ran successfully with the following output:
----
--output tsv
----

✗ condition not met, skipping pipeline
```

=> `az login required`  means that the `az` command is not logged-in: other environment variables are required

File: Jenkinsfile_updatecli [hunk 1]
```diff
@@ -21,13 +21,27 @@ pipeline {
...
         catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
-          updatecli(action: 'diff', cronTriggerExpression: cronExpr)
+          script {
+            withCredentials([
+              usernamePassword(
+                credentialsId: 'github-app-updatecli-on-jenkins-infra',
+                usernameVariable: 'USERNAME_VALUE', // Setting this variable is mandatory, even if of not used when the credentials is a githubApp one
+                passwordVariable: 'UPDATECLI_GITHUB_TOKEN'
+              )
+            ]) {
```
Comment (dduportal) [inline Jenkinsfile_updatecli:38]: as discussed together, these instructions duplicates the 
```
UPDATECLI_AZURE = credentials('updatecli-azure-serviceprincipal')
```

instruction

File: Jenkinsfile_updatecli [hunk 2]
```diff
@@ -36,7 +50,21 @@ pipeline {
...
+              usernamePassword(
+                credentialsId: 'github-app-updatecli-on-jenkins-infra', //needed for updatecli
+                usernameVariable: 'USERNAME_VALUE', // Setting this variable is mandatory, even if of not used when the credentials is a githubApp one
+                passwordVariable: 'UPDATECLI_GITHUB_TOKEN'
+              )
+            ]) {
+              sh 'az login --service-principal -u "$UPDATECLI_AZURE_CLIENT_ID" -p "$UPDATECLI_AZURE_CLIENT_SECRET" -t "$UPDATECLI_AZURE_TENANT_ID"'
+              sh 'az account set -s "$UPDATECLI_AZURE_SUBSCRIPTION_ID"'
+              sh 'updatecli version'
+              sh 'updatecli diff --values ./updatecli/values.yaml --config ./updatecli/updatecli.d'
```
Comment (dduportal) [inline Jenkinsfile_updatecli:64]:
```suggestion
```

File: Jenkinsfile_updatecli (hunk 1 above)
Comment (smerle33) [reply Jenkinsfile_updatecli]: as seen together, not as simple, we still need both for now.

File: Jenkinsfile_updatecli (hunk 2 above)
Comment (smerle33) [reply Jenkinsfile_updatecli:64]: yes, useless, thanks to get this !

**Review by dduportal — APPROVED** [review]
```
```
[2023-05-25T08:14:50.019Z] CONDITIONS:
[2023-05-25T08:14:50.019Z] ===========
[2023-05-25T08:14:50.019Z] 
[2023-05-25T08:14:50.019Z] checkifazureimagejenkins-agent-ubuntu-22.04-arm64isavailable
[2023-05-25T08:14:50.020Z] ------------------------------------------------------------
[2023-05-25T08:14:51.391Z] The shell 🐚 command "/bin/sh /tmp/updatecli/bin/ed4f9b44a7b060335370c9eed7ab990aa76fae61a99d298eaf9fa07597e4a3c4.sh" ran successfully with the following output:
[2023-05-25T08:14:51.392Z] ----
[2023-05-25T08:14:51.392Z] /subscriptions/****/resourceGroups/prod-packer-images/providers/Microsoft.Compute/galleries/prod_packer_images/images/jenkins-agent-ubuntu-22.04-arm64/versions/1.7.1	eastus2	1.7.1	Succeeded		None	prod-packer-images				Microsoft.Compute/galleries/images/versions
[2023-05-25T08:14:51.392Z] ----
[2023-05-25T08:14:51.392Z] 
[2023-05-25T08:14:51.392Z] checkifazureimagejenkins-agent-windows-2019-amd64isavailable
[2023-05-25T08:14:51.392Z] ------------------------------------------------------------
[2023-05-25T08:14:51.955Z] The shell 🐚 command "/bin/sh /tmp/updatecli/bin/1ea704c3494035bf5129962ef4699267e46009bc74a98de9fd563b539af186fc.sh" ran successfully with the following output:
[2023-05-25T08:14:51.955Z] ----
[2023-05-25T08:14:51.955Z] /subscriptions/****/resourceGroups/prod-packer-images/providers/Microsoft.Compute/galleries/prod_packer_images/images/jenkins-agent-windows-2019-amd64/versions/1.7.1	eastus2	1.7.1	Succeeded		None	prod-packer-images				Microsoft.Compute/galleries/images/versions
[2023-05-25T08:14:51.955Z] ----
[2023-05-25T08:14:51.955Z] 
[2023-05-25T08:14:51.955Z] checkifazureimagejenkins-agent-ubuntu-22.04-amd64isavailable
[2023-05-25T08:14:51.955Z] ------------------------------------------------------------
[2023-05-25T08:14:52.885Z] The shell 🐚 command "/bin/sh /tmp/updatecli/bin/a960da1e632bc8415ec18a2d9030b97658fff153e64589f150d7095ff0a5747e.sh" ran successfully with the following output:
[2023-05-25T08:14:52.886Z] ----
[2023-05-25T08:14:52.886Z] /subscriptions/****/resourceGroups/prod-packer-images/providers/Microsoft.Compute/galleries/prod_packer_images/images/jenkins-agent-ubuntu-22.04-amd64/versions/1.7.1	eastus2	1.7.1	Succeeded		None	prod-packer-images				Microsoft.Compute/galleries/images/versions
[2023-05-25T08:14:52.886Z] ----
[2023-05-25T08:14:52.886Z] 
[2023-05-25T08:14:52.886Z] 
[2023-05-25T08:14:52.886Z] TARGETS
[2023-05-25T08:14:52.886Z] ========
[2023-05-25T08:14:52.886Z] 
[2023-05-25T08:14:52.886Z] setAzureGalleryImageVersion
[2023-05-25T08:14:52.886Z] ---------------------------
[2023-05-25T08:14:52.886Z] 
[2023-05-25T08:14:52.886Z] **Dry Run enabled**
[2023-05-25T08:14:52.886Z] 
[2023-05-25T08:14:53.451Z] ✔ Content from file "/tmp/updatecli/github/jenkins-infra/kubernetes-management/config/jenkins_infra.ci.jenkins.io.yaml" already up to date
[2023-05-25T08:14:53.451Z] ✔ All contents from 'file' and 'files' combined already up to date
```

LGTM 🚀

**Pull request merged**